R/vars_funs.R

```diff
@@ -123,33 +62,47 @@ vars_check_class <- function(age, sqft, class) {
 #'   data = sample_data_athena,
 #'   names_from = "athena",
 #'   names_to = "model",
-#'   dict = ccao::vars_dict
+#'   dictionary = ccao::vars_dict
 #' )
 #' vars_rename(
 #'   data = sample_data_athena,
 #'   names_from = "athena",
 #'   names_to = "pretty",
-#'   dict = ccao::vars_dict
+#'   dictionary = ccao::vars_dict
 #' )
 #' @md
 #' @family vars_funs
 #' @export
 vars_rename <- function(data,
                         names_from = NULL,
                         names_to = NULL,
-                        type = "inplace",
-                        dict = ccao::vars_dict) {
+                        output_type = "inplace",
+                        dictionary = ccao::vars_dict,
+                        # Deprecated args
+                        type = NULL,
+                        dict = NULL) {
+  # Check if deprecated arguments are used and override values if so
+  if (!is.null(type)) {
+    warning("'type' is deprecated. Use 'output_type' instead.", call. = FALSE)
+    output_type <- type
+  }
+
+  if (!is.null(dict)) {
+    warning("'dict' is deprecated. Use 'dictionary' instead.", call. = FALSE)
+    dictionary <- dict
+  }
```

Comment:
**suggestion (non-blocking):** I recognize this is good practice, but I wonder about the utility of including lifecycle code like this in a strictly internal package. My preference would be to just release a new major version with the breaking changes included. Ultimately I defer to you @jeancochrane, as my view here isn't super strongly held.

Reply:
Makes sense, I stripped out this logic in 3d94d7e7e140851fc1afb9a313a7e9828ef4c905.